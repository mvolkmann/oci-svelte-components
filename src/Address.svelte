<script>
  //import {get} from 'lodash';
  import get from 'lodash/get';
  import LabeledCountrySelect from './LabeledCountrySelect.svelte';
  import LabeledInput from './LabeledInput.svelte';
  import LabeledStateSelect from './LabeledStateSelect.svelte';
  import {globalStore, setState} from './stores';
  console.log('Address.svelte x: globalStore =', globalStore);

  export let statePath;
  export let vertical = false;

  let countryCode = '';

  $: countryCode = get ? get($globalStore, statePath + '.country') : '';

  $: zipLabel =
    countryCode === 'USA' ? 'Zip' : countryCode === 'CAN' ? 'Postal Code' : '';

  const clearState = () => setState(statePath + '.state', '');
</script>

<div class="address">
  <LabeledCountrySelect
    on:change={clearState}
    statePath={statePath + '.country'}
    {vertical} />
  {#if countryCode}
    <LabeledInput label="Street" statePath={statePath + '.street'} {vertical} />
    <LabeledInput label="City" statePath={statePath + '.city'} {vertical} />
    <LabeledStateSelect
      {countryCode}
      statePath={statePath + '.state'}
      {vertical} />
    {#if zipLabel}
      <LabeledInput
        label={zipLabel}
        statePath={statePath + '.zip'}
        {vertical} />
    {/if}
  {/if}
</div>
