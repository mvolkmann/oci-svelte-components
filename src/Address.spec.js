import {cleanup, render} from '@testing-library/svelte';

import Address from './Address.svelte';

describe('Address', () => {
  afterEach(cleanup);

  test('should match snapshot', () => {
    const path = 'user.profile.address';
    const {container} = render(Address, {path});
    expect(container).toMatchSnapshot();
  });
});
