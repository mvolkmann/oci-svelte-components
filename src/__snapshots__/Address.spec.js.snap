// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address should match snapshot 1`] = `
<body>
  <div>
    <div
      class="address"
    >
      <div
        class="osc-labeled svelte-1tulwxk"
      >
        <label
          class="svelte-74lyeh"
          for=""
        >
          Country
           
           
        </label>
         
        <div
          class="container svelte-seedhc"
        >
          <select
            class="osc-labeled-select svelte-seedhc"
          >
            <option
              value=""
            >
              
            </option>
            <option
              value="AUS"
            >
              Australia
               
            </option>
            <option
              value="BRA"
            >
              Brazil
               
            </option>
            <option
              value="CAN"
            >
              Canada
               
            </option>
            <option
              value="CHN"
            >
              China
               
            </option>
            <option
              value="FRA"
            >
              France
               
            </option>
            <option
              value="DEU"
            >
              Germany
               
            </option>
            <option
              value="IND"
            >
              India
               
            </option>
            <option
              value="ITA"
            >
              Italy
               
            </option>
            <option
              value="JPN"
            >
              Japan
               
            </option>
            <option
              value="MEX"
            >
              Mexico
               
            </option>
            <option
              value="NOR"
            >
              Norway
               
            </option>
            <option
              value="POL"
            >
              Poland
               
            </option>
            <option
              value="RUS"
            >
              Russia
               
            </option>
            <option
              value="ESP"
            >
              Spain
               
            </option>
            <option
              value="SWE"
            >
              Sweden
               
            </option>
            <option
              value="CHE"
            >
              Switzerland
               
            </option>
            <option
              value="GBR"
            >
              United Kingdom
               
            </option>
            <option
              value="USA"
            >
              United States of America
               
            </option>
          </select>
        </div>
         
      </div>
       
    </div>
  </div>
</body>
`;
